src/apache.c

```diff
@@ -586,14 +587,30 @@ static int apache_init (void) /* {{{ */
 {
 	/* Call this while collectd is still single-threaded to avoid
 	 * initialization issues in libgcrypt. */
+	if(cd_ssl_multithread_setup() == SSL_MT_OK){
```

Comment:
If I understand the problem correctly, this needs to be called from every plugin using libcurl. I'd therefore try to keep this as short as possible, i.e. just `cd_ssl_multithread_setup()` — the function will print a warning itself and since we don't abort, we don't change the behavior by ignoring the return value.